ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/ProcessingLoggerFactory.java

```diff
@@ -19,11 +19,21 @@
 
 public interface ProcessingLoggerFactory {
   /**
-   * Get a processing logger for writing record processing log messages
+   * Get a processing logger for writing record processing log messages. This processing logger
+   * also emits metrics
    * @param name The name of the logger to get.
-   * @return The logger with the given name.
+   * @return The logger with the given name that emits metrics.
    */
-  ProcessingLogger getLogger(String name);
+  ProcessingLogger getLoggerWithMetrics(String name);
+
+  /**
+   * Get a processing logger for writing record processing log messages. This processing logger
+   * also emits metrics
+   * @param name The name of the logger to get.
+   * @param queryId The queryId to tag the metrics with
+   * @return The logger with the given name that emits metrics.
+   */
+  ProcessingLogger getLoggerWithMetrics(String name, String queryId);
```

Comment:
Lets leave the queryId out of this interface and make it more generic, like:

```
ProcessingLogger getLoggerWithMetrics(String name, Map<String, String> additionalMetricTags);
```

Reply:
changed it to take a map